### DISCUSSION TITLE
Replace `ffi` with `stdJson` in `constructCreate2Salt` function

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/evm-monorepo#904] [Ideas]
During our call with Rusty (from Cantina), he suggested us to avoid using `ffi` for the following reasons:

- Enabling `ffi` allows executing any arbitrary command during foundry tests. This is what enabled us to use `jq` cli to read value of `version` from `package.json`
- "It is generally advised to use this cheat code as a last resort, and to not enable it by default, as anyone who can change the tests of a project will be able to execute arbitrary commands on devices that run the tests" - [Foundry doc](https://book.getfoundry.sh/cheatcodes/ffi#ffi)

However, a safer solution is to use [`stdJson`](https://book.getfoundry.sh/cheatcodes/parse-json?highlight=stdJson#how-to-use-stdjson) which allows us to read `package.json` as a JSON object. This is an in-build cheat code in Foundry and doesn't require us to enable `ffi`. It also reduces our dependency from `jq` cli and reduces the number of lines required to create salt.

The following function:
```solidity
function constructCreate2Salt() public returns (bytes32) {
    string memory chainId = block.chainid.toString();
    string[] memory inputs = new string[](4);
    inputs[0] = "jq";
    inputs[1] = "-r";
    inputs[2] = ".version";
    inputs[3] = "./package.json";
    bytes memory result = vm.ffi(inputs);
    string memory version = string(result);
    string memory create2Salt = string.concat("ChainID ", chainId, ", Version ", version);
    console2.log("The CREATE2 salt is \"%s\"", create2Salt);
    return bytes32(abi.encodePacked(create2Salt));
}
```
becomes

```solidity
function constructCreate2Salt() public view returns (bytes32) {
    string memory chainId = block.chainid.toString();
    string memory json = vm.readFile("package.json");
    string memory version = json.readString(".version");
    string memory create2Salt = string.concat("ChainID ", chainId, ", Version ", version);
    console2.log("The CREATE2 salt is \"%s\"", create2Salt);
    return bytes32(abi.encodePacked(create2Salt));
}
```

I've already implemented it in https://github.com/sablier-labs/v2-core/commit/2cc42e51793df5035bbdb1b81818b1a6171a6e06 if we agree to go ahead with this.

**Comment by PaulRBerg** [comment]
This sounds like low-risk to me but happy to accept the change.

**Comment by smol-ninja** [comment]
Related PRs:
- https://github.com/sablier-labs/v2-core/pull/907
- https://github.com/sablier-labs/v2-periphery/pull/333
